props/saber_BC_buttons.h

```diff
@@ -383,34 +1158,253 @@ EFFECT(voldown);    // for decrease volume
+    float volume = dynamic_mixer.get_volume();
+    percentage_ = round((volume / max_volume_) * 10) * 10;
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    // Overridden to substitute the tick sound
+    this->say_time_ = Cyclint<uint32_t>(millis()) + 1;
+    if (!this->say_time_) this->say_time_ += 1;
+  }
+
+  void select() override {
+    PVLOG_NORMAL << "** Saved - Exit Volume Menu\n";
+    mode::getSL<SPEC>()->SaySave();
+    SPEC::SteppedMode::select();
+  }
+
+  void exit() override {
+    PVLOG_NORMAL << "** Cancelled - Exit Volume Menu\n";
+    percentage_ = initial_percentage_;
+    dynamic_mixer.set_volume(initial_volume_);
+    mode::getSL<SPEC>()->SayCancel();
+    SPEC::SteppedMode::exit();
+  }
+
+  bool mode_Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, 0)) {
+      // Custom button controls for BCVolumeMode
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_HELD_MEDIUM, 0):
+        QuickMaxVolume();
+        return true;
+
+      case EVENTID(BUTTON_POWER, EVENT_SECOND_HELD_MEDIUM, 0):
+      case EVENTID(BUTTON_AUX, EVENT_FIRST_HELD_MEDIUM, 0):
+        QuickMinVolume();
+        return true;
     }
+    // Use the select and exit controls from SelectCancelMode
+    return SPEC::SelectCancelMode::mode_Event2(button, event, modifiers);
+  }
+};
+
+#ifdef DYNAMIC_BLADE_LENGTH
+
+template<class SPEC>
+struct BCSelectBladeMode : public SPEC::MenuBase {
+  int current_blade() { return this->pos_ + 1; }
+  ShowColorSingleBladeTemplate<Pulsing<ShowColorStyle,Black,800>,Pulsing<ShowColorStyle,Black,800>> highlighted_blade_;
+
+  uint16_t size() override {
+    return NUM_BLADES;
+  }
+  void mode_activate(bool onreturn) override {
+    mode::getSL<SPEC>()->SaySelectBlade();
+    ShowColorStyle::SetColor(Color16(65535, 65535, 65535));
+    SPEC::SteppedMode::mode_activate(onreturn);
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+    // say();
+  }
+
+  void mode_deactivate() {
+    highlighted_blade_.Stop(current_blade());
+  }
+
+  void next() override {
+    highlighted_blade_.Stop(current_blade());
+    SPEC::MenuBase::next();
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+  }
+
+  void prev() override {
+    highlighted_blade_.Stop(current_blade());
+    SPEC::MenuBase::prev();
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+  }
+
+  void say() override {
+    mode::getSL<SPEC>()->SayBlade();
+    mode::getSL<SPEC>()->SayWhole(current_blade());
+  }
+
+  void select() override {
+    // Set the current blade to send and push to ChangeBladeLengthMode
+    mode::menu_current_blade = current_blade();
+    highlighted_blade_.Stop(current_blade());
+    pushMode<typename SPEC::ChangeBladeLengthMode>();
+  }
+
+  void exit() override {
```

Comment:
An overridden function that just calls the inherited base function can be omitted.

Reply:
Removed.